airflow/providers/google/cloud/transfers/bigquery_to_mssql.py

```diff
@@ -39,86 +38,49 @@ class BigQueryToMsSqlOperator(BaseOperator):
-    :param database: name of database which overwrite defined one in connection
-    :param replace: Whether to replace instead of insert
-    :param batch_size: The number of rows to take in each batch
-    :param location: The location used for the operation.
-    :param impersonation_chain: Optional service account to impersonate using short-term
-        credentials, or chained list of accounts required to get the access_token
-        of the last account in the list, which will be impersonated in the request.
-        If set as a string, the account must grant the originating account
-        the Service Account Token Creator IAM role.
-        If set as a sequence, the identities from the list must grant
-        Service Account Token Creator IAM role to the directly preceding identity, with first
-        account from the list granting this role to the originating account (templated).
     """
 
-    template_fields: Sequence[str] = ("source_project_dataset_table", "mssql_table", "impersonation_chain")
+    template_fields: Sequence[str] = tuple(BigQueryToSqlBaseOperator.template_fields) + (
+        "source_project_dataset_table",
+    )
     operator_extra_links = (BigQueryTableLink(),)
 
     def __init__(
         self,
         *,
         source_project_dataset_table: str,
-        mssql_table: str,
-        selected_fields: list[str] | str | None = None,
-        gcp_conn_id: str = "google_cloud_default",
+        mssql_table: str | None = None,
+        target_table_name: str | None = None,
         mssql_conn_id: str = "mssql_default",
-        database: str | None = None,
-        replace: bool = False,
-        batch_size: int = 1000,
-        location: str | None = None,
-        impersonation_chain: str | Sequence[str] | None = None,
         **kwargs,
     ) -> None:
-        super().__init__(**kwargs)
-        self.selected_fields = selected_fields
-        self.gcp_conn_id = gcp_conn_id
-        self.mssql_conn_id = mssql_conn_id
-        self.database = database
-        self.mssql_table = mssql_table
-        self.replace = replace
-        self.batch_size = batch_size
-        self.location = location
-        self.impersonation_chain = impersonation_chain
+        if mssql_table is not None:
+            warnings.warn(
+                "The `mssql_table` parameter has been deprecated. Use `target_table_name` instead.",
+                DeprecationWarning,
+            )
+
         try:
-            _, self.dataset_id, self.table_id = source_project_dataset_table.split(".")
+            _, dataset_id, table_id = source_project_dataset_table.split(".")
         except ValueError:
             raise ValueError(
                 f"Could not parse {source_project_dataset_table} as <project>.<dataset>.<table>"
             ) from None
+        super().__init__(
+            target_table_name=target_table_name or mssql_table, dataset_table=f"{dataset_id}.{table_id}", **kwargs
```

Comment:
lets not do `target_table_name or mssql_table`
it makes it harder to understand where the "real" value is.
we should handle the value as part of the deprecation check. If user passed `mssql_table` then we also assign it to `self. target_table_name`

Reply:
I'm not sure to understand what you have in mind. Something like this?

```
    selected_target_table_name = target_table_name

    if mssql_table is not None:
            selected_target_table_name = target_table_name
            warnings.warn(
                "The `mssql_table` parameter has been deprecated. Use `target_table_name` instead.",
                DeprecationWarning,
            )

        super().__init__(
            target_table_name=selected_target_table_name,
            dataset_table=f"{dataset_id}.{table_id}",
            **kwargs,
        )
```

Reply:
Something like https://github.com/apache/airflow/blob/23b5b31d48a1c8fc784022c641b33b4ba837d549/airflow/providers/slack/operators/slack.py#L210-L221
basically once you pass that block 

New parameter takes precedence but if older parameter is supplied we will assign it to the new parameter + raise warning.